mappings/net/minecraft/recipe/RecipeMatcher.mapping

```diff
@@ -0,0 +1,120 @@
+CLASS net/minecraft/class_1662 net/minecraft/recipe/RecipeMatcher
+	COMMENT Matching class that matches a recipe to its required resources.
+	COMMENT This specifically does not check patterns (See {@link ShapedRecipe} for that.)
+	FIELD field_7550 inputs Lit/unimi/dsi/fastutil/ints/Int2IntMap;
+	METHOD method_20478 addInput (Lnet/minecraft/class_1799;I)V
+		COMMENT Adds an item stack to the pool of available resources.
+		ARG 1 stack
+		ARG 2 maxCount
+	METHOD method_7400 addInput (Lnet/minecraft/class_1799;)V
+		COMMENT Adds a full item stack to the pool of available resources.
+		COMMENT
+		COMMENT <p>This is equivalent to calling {@code addInput(stack, ItemStack.MAX_COUNT)}</p>
+		ARG 1 stack
+	METHOD method_7401 addInput (II)V
+		COMMENT Adds an input to be used for recipe matching.
+		ARG 1 itemId
+			COMMENT The raw id of the item to match.
+		ARG 2 count
+			COMMENT The item count. Number of instances.
+	METHOD method_7402 match (Lnet/minecraft/class_1860;Lit/unimi/dsi/fastutil/ints/IntList;)Z
+		COMMENT Attempts to match the recipe against the collected inputs.
+		COMMENT Assumes only one output is required.
+		ARG 1 recipe
+			COMMENT The recipe to match against
+		ARG 2 output
+			COMMENT Optional output list of item ids that were matched whilst evaulating the recipe conditions.
```

Comment:
What does "optional" even mean here? Does it mean the list can be `null` or have `size() == 0`?

Reply:
I can't remember exactly, but I think it is @Nullable